### PR TITLE
Add basic `retry()` and `repeat()` support to generated tests.

### PR DESCRIPTION
**Pull request by PCMan**
When generating tests, if the `test_logic` function is decorated with
either `retry()` or `repeat()` decorators, apply the settings to the
generated test methods.

---
This change is [Reviewable](https://reviewable.io/reviews/google/mobly/766)

### CODE REVIEW

**Review by eric100lin — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly/766)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @PCMan and @xpconanfan)

___
*[tests/mobly/base_test_test.py, line 2484 at r1](https://reviewable.io/reviews/google/mobly/766#-Ml9Ic1Z1UX0SbCbhuvF:-Ml9Ic1Z1UX0SbCbhuvG:bh3i5tu) ([raw file](https://github.com/google/mobly/blob/205d504a4e94425249f5ced67fa6d76e63cd6ad2/tests/mobly/base_test_test.py#L2484)):*
> ```
>     mock_action = mock.MagicMock()
>     mock_action.side_effect = [Exception('Fail 1'), Exception('Fail 2'), None]
> ```
nit: these two lines can be merged into one line:
```
    mock_action = mock.MagicMock(
      side_effect=[Exception('Fail 1'), Exception('Fail 2'), None])
```

**Review by eric100lin — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/google/mobly/766#-:-Ml9ItKj6K_GX4IFDf1r:bnfp4nl)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @PCMan and @xpconanfan)

File: mobly/base_test.py [hunk 1]
```diff
@@ -862,6 +862,11 @@ def generate_tests(self, test_logic, name_func, arg_sets, uid_func=None):
         raise Error('%s Test name "%s" already exists, cannot be duplicated!' %
                     (root_msg, test_name))
       test_func = functools.partial(test_logic, *args)
+      for attr_name in (
```
Comment (xpconanfan) [inline mobly/base_test.py:865]: we are essentially applying the decoration attrs to the generated func right?
can you add an inline comment explaining what this is doing so it's easier to understand for ppl without the context?

**Review by xpconanfan — CHANGES_REQUESTED** [review]

File: mobly/base_test.py (hunk 1 above)
Comment (PCMan) [reply mobly/base_test.py:869]: Comments added! Thanks!

**Review by eric100lin — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/google/mobly/766#-:-Ml9Q50w8LqtTAkSzBVh:bnfp4nl)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @PCMan and @xpconanfan)

**Review by eric100lin — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/google/mobly/766#-:-MpjDyLn1XAPgyM1o5xu:bnfp4nl)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @xpconanfan)

**Review by xpconanfan — APPROVED** [review]

**Pull request merged**